h/schemas/api/group.py

```diff
@@ -24,8 +25,70 @@ class CreateGroupAPISchema(JSONSchema):
                 'type': 'string',
                 'maxLength': GROUP_DESCRIPTION_MAX_LENGTH,
             },
+            'groupid': {
+                'type': 'string',
+                'pattern': GROUPID_PATTERN,
+            },
         },
         'required': [
             'name'
         ],
     }
+
+    def __init__(self, group_authority=None, default_authority=None):
+        """
+        The ``group_authority`` and ``default_authority`` properties are used
+        to validate a ``groupid`` field in data. If the caller does not intend
+        to pass any data dicts that contain a ``groupid`` entry, these
+        properties are optional.
+
+        :param group_authority:   The authority that is be associated with the group
+        :param default_authority: The service's default authority; if it is the same as
+                                  ``group_authority``, then this is considered a
+                                  "first-party" group
+        """
+        super(CreateGroupAPISchema, self).__init__()
+        self.group_authority = group_authority
+        self.default_authority = default_authority
+
+    def validate(self, data):
+        appstruct = super(CreateGroupAPISchema, self).validate(data)
+        groupid_parts = self._validate_groupid(appstruct)
+        if groupid_parts is not None:
+            appstruct['authority_provided_id'] = groupid_parts['authority_provided_id']
```

Comment:
I hope it's not too magical to have something appear on the `appstruct` that wasn't there before? I feel like our form processing does this but it may be funky on a JSON Schema class.